Dockerfile.windows-build

```diff
@@ -41,24 +42,49 @@ RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --de
 ENV PATH="/root/.cargo/bin:${PATH}"
 RUN rustup target add x86_64-pc-windows-gnullvm
 
+# Configure Cargo to use clang as linker for Windows target
+RUN mkdir -p /root/.cargo && \
+    printf '[target.x86_64-pc-windows-gnullvm]\n\
+linker = "clang"\n\
+rustflags = [\n\
+    "-C", "link-arg=--target=x86_64-w64-mingw32",\n\
+    "-C", "link-arg=--sysroot=/clang64",\n\
+    "-C", "link-arg=-fuse-ld=lld",\n\
+    "-C", "link-arg=--rtlib=compiler-rt",\n\
```

Comment:
I'm very surprised we would need to add compiler-rt and force the use of lld here, but if it works, it works I suppose